### DISCUSSION TITLE
Shall I use unlocked or managed package?

### DISCUSSION THREAD

**Discussion by reitffunk** [jongpie/NebulaLogger#282] [Q&A] [answered]
Hi,

basically what I got is that the managed package is more easily to update. Also it does not clutter my IDE with classes from the logger what I personally prefer. I do not want to change logic of external packages as it seems counterproductive to me.

The only issue I see is the version difference then. Currently the unlocked package is at version 4.6.16 but the managed package only at version 4.6.0.

Does that mean I miss all the bug fixes from 4.6.0 to 4.6.16 while using a managed package? Or how could I get the latest fixes in the managed package?

Regards!

**Comment by jongpie** [comment, marked as answer]
Hi @reitffunk - I _highly_ recommend that you use the unlocked package, and not the managed package. I have [a page in the wiki](https://github.com/jongpie/NebulaLogger/wiki/Unlocked-vs-Managed-Package) with some additional details, but essentially:

- The unlocked package is much more powerful overall - some features do not work in the managed package due to some limitations with managed packages
- Upgrading the unlocked package in your org is actually more reliable than upgrading the managed package. Another limitation with managed packages is that some metadata, like some layout changes, new picklist values, etc., are not propagated to your org when upgrading managed packages. Unlocked packages provide a better upgrade experience.
- The unlocked package (as you noted) receives much more frequent updates - this includes bugfixes, as well as new features. The managed package will only have 3 planned releases this calendar year (to mimic Salesforce's 3 releases for Spring, Summer & Winter)
- If you're using `sfdx` within your IDE, then you should be fine - by default, `sfdx` will not retrieve metadata in your org if it's part of a package, so you shouldn't get any of Nebula Logger's metadata when using either the unlocked or managed package

Hope this helps, but let me know if you have any follow-up questions!

**Reply by reitffunk** [reply]
Thank you for the detailed answer, explains it very well!

Next time I will have a deeper look into the wiki first. :)

**Reply by jongpie** [reply]
Excellent, glad it was helpful! Feel free to reach out again if you need anything else!
